File: BestTimeToBuyAndSellStock/WithCooldown/BestTimetoBuyandSellStockwithCooldown.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) <= 1:
            return 0
        buy = [- prices[0]]
        sell = [0]
        for i in range(1, len(prices)):
            buy += max(sell[i - 2] - prices[i], buy[i - 1]),
            sell += max(buy[i - 1] + prices[i], sell[i - 1]),
        return sell[len(prices) - 1]

File: BestTimeToBuyAndSellStock/WithCooldown/BestTimetoBuyandSellStockwithCooldown.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
		if(prices.size() < 1)
			return 0;
		vector<int> buy = {-prices[0]};
		vector<int> sell = {prices[0]};
		for(int i = 1;i < prices.size(); i++) {
			buy.push_back(max(sell[i - 2] - prices[i], buy[i - 1]));
			sell.push_back(max(buy[i - 1] + prices[i], sell[i - 1]));
		}
		return sell[prices.size() - 1];
    }
};